### PR TITLE
Add maximum constraint on atdgen-runtime

### PR DESCRIPTION
**Pull request by Leonidas-from-XIV**
Lots of packages failing on the new Yojson are due to atdgen-runtime using the Yojson 1.x Biniou API, which was removed in 2.x in favour of using Buffer.

This should solve a lot of failing reverse dependencies.

### CODE REVIEW

File: packages/atdgen-runtime/atdgen-runtime.2.0.0/opam [hunk 1]
```diff
@@ -20,7 +20,7 @@ depends: [
   "ocaml" {>= "4.02.3"}
   "jbuilder" {>= "1.0+beta7"}
```
Comment (mseri) [inline packages/atdgen-runtime/atdgen-runtime.2.0.0/opam:21]:
```suggestion
  "jbuilder" {>= "1.0+beta9"}
```

jbuilder fails with the [Unknown field package](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/97a35ce36bbc5a62adb912c1ef5feab46231a846/variant/opam-2.1,compilers,4.07,atdgen-runtime.2.0.0,lower-bounds) error

File: packages/atdgen-runtime/atdgen-runtime.1.13.0/opam [hunk 2]
```diff
@@ -14,7 +14,7 @@ depends: [
   "ocaml" {>= "4.02.3"}
   "jbuilder" {>= "1.0+beta7"}
```
Comment (mseri) [inline packages/atdgen-runtime/atdgen-runtime.1.13.0/opam:15]:
```suggestion
  "jbuilder" {>= "1.0+beta9"}
```

File: packages/atdgen-runtime/atdgen-runtime.1.13.0/opam [hunk 3]
```diff
@@ -12,9 +12,9 @@ build: [
 
 depends: [
   "ocaml" {>= "4.02.3"}
-  "jbuilder" {>= "1.0+beta7"}
+  "jbuilder" {>= "1.0+beta9"}
```
Comment (Leonidas-from-XIV) [inline packages/atdgen-runtime/atdgen-runtime.1.13.0/opam:15]:
```suggestion
  "jbuilder" {>= "1.0+beta17"}
```
Accodring to the [changelog](https://github.com/ocaml/dune/blob/main/CHANGES.md#10beta17-01022018=) `include` was added in beta17.

File: packages/atdgen-runtime/atdgen-runtime.2.0.0/opam [hunk 4]
```diff
@@ -18,9 +18,9 @@ build: [
 
 depends: [
   "ocaml" {>= "4.02.3"}
-  "jbuilder" {>= "1.0+beta7"}
+  "jbuilder" {>= "1.0+beta9"}
```
Comment (Leonidas-from-XIV) [inline packages/atdgen-runtime/atdgen-runtime.2.0.0/opam:21]:
```suggestion
  "jbuilder" {>= "1.0+beta17"}
```
According to the [changelog](https://github.com/ocaml/dune/blob/main/CHANGES.md#10beta17-01022018=) `diff` was added in beta17.

**Comment by mseri** [comment]
I have also added Atdgen lower bounds for all those failures `Error: Unbound value Atdgen_runtime.Oj_run.missing_field`, hopefully I caught them all

File: packages/atdgen-runtime/atdgen-runtime.2.0.0/opam [hunk 5]
```diff
@@ -18,9 +18,9 @@ build: [
 
 depends: [
   "ocaml" {>= "4.02.3"}
-  "jbuilder" {>= "1.0+beta7"}
+  "jbuilder" {>= "1.0+beta17"}
   "biniou" {>= "1.0.6"}
-  "yojson" {>= "1.2.1"}
+  "yojson" {>= "1.2.1" & < "2.0.0"}
```
Comment (mseri) [inline packages/atdgen-runtime/atdgen-runtime.2.0.0/opam:23]:
```suggestion
  "yojson" {>= "1.3.1" & < "2.0.0"}
```
Due to [Unbound module Yojson.Safe.Util)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/3c9a30ea33444c325db73bbd0f999c4dac03d37a/variant/opam-2.1,compilers,4.03,atdgen-runtime.2.0.0,lower-bounds)

Comment (Leonidas-from-XIV) [inline packages/atdgen-runtime/atdgen-runtime.2.0.0/opam]:
```suggestion
  "yojson" {>= "1.7.0" & < "2.0.0"}
```

1.2.1 is missing `Yojson.Safe.Util` but I am not entirely sure when this was added (it is not mentioned in the changelog). Given all the other atdgen-runtime versions use `1.7.0` as minimum constraint I feel like that's probably an ok boundary.

Comment (mseri) [reply packages/atdgen-runtime/atdgen-runtime.2.0.0/opam]: I looked in the sources, it was introduced in 1.3.1

**Comment by mseri** [comment]
Oh no. Looks like atdgen and atdgen-runtime are  version-decoupled 😓

**Comment by mseri** [comment]
The last commit seems to have worked. Thanks for the help

**Pull request merged**